httpsync/publisher.go

```diff
@@ -93,13 +92,20 @@ func (p *publisher) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		// serve the
 		p.rl.RLock()
 		defer p.rl.RUnlock()
-		out, err := json.Marshal(p.root.String())
+
+		if p.privKey == nil {
+			http.Error(w, "Server not initialized", http.StatusInternalServerError)
+			log.Errorw("Failed to load private key to sign head request")
+			return
+		}
```

Comment:
I think this check should be moved to `NewPublisher`, so that a publisher cannot even be created without proper initialization.